core/client/client.go

```diff
@@ -124,6 +124,13 @@ func New(ktClient pb.KeyTransparencyClient,
 	}
 }
 
+func (c *Client) updateTrusted(newTrusted *trillian.SignedLogRoot) {
+	if newTrusted.TimestampNanos > c.trusted.TimestampNanos &&
+		newTrusted.TreeSize >= c.trusted.TreeSize {
+		c.trusted = *newTrusted
```

Comment:
If you get a newer root with a smaller tree size that's possibly a bad thing that should maybe be exposed.

Reply:
Hmm, that plays into the server skew question from yesterday. 
If we get an older root than the currently trusted root, the consistency proof will fail.